linkml/generators/shacl/shacl_ifabsent_processor.py

```diff
@@ -62,12 +62,29 @@ def map_datetime_default_value(
         return Literal(f"{year}-{month}-{day}T{hour}:{minutes}:{seconds}", datatype=ShaclDataType.DATETIME.uri_ref)
 
     def map_uri_or_curie_default_value(self, default_value: str, slot: SlotDefinition, cls: ClassDefinition):
-        raise NotImplementedError()
+        is_curie = False
+        if default_value in self.CURIE_SPECIAL_CASES:
+            is_curie = True
+            default_value = self._map_curie_special_case(default_value, slot, cls)
+        elif default_value in self.URI_SPECIAL_CASES:
+            default_value = self._map_uri_special_case(default_value, slot, cls)
+        else:
+            if ":" in default_value and "://" not in default_value and len(default_value.split(":")) == 2:
```

Comment:
"Been there, seen that" 😄 

I introduced a module for much more robust URI/CURIE validation some time ago: https://github.com/linkml/linkml-runtime/blob/main/linkml_runtime/utils/uri_validator.py You can simply use `validate_curie`.

Although, on the other side, if you can simply assume that the CURIEs/URIs are valid, your logic is simpler.

Reply:
Oh great, I was looking for this and didnt know where to find it